python/ray/serve/pipeline/tests/test_pipeline.py

```diff
@@ -97,6 +96,9 @@ def test_class_constructor_not_called_until_deployed(execution_mode,
 
     with tempfile.NamedTemporaryFile("w") as tmp:
 
+        tmp_name = tmp.name + ".tmp"
+        open(tmp_name, "w+").close()
+
```

Comment:
Without this and the following changes I get following permission errors because (probably) Windows confuses `C:\\CONDA_~1\\tmp04t9xjnp` with a directory.

After this change the test passes in all execution modes.

```bash
            file_writer = FileWriter(tmp.name, msg)
>           assert not constructor_called()

python\ray\serve\pipeline\tests\test_pipeline.py:118:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

    def constructor_called():
>       with open(tmp.name, "r") as f:
E       PermissionError: [Errno 13] Permission denied: 'C:\\CONDA_~1\\tmp04t9xjnp'

python\ray\serve\pipeline\tests\test_pipeline.py:114: PermissionError
```